test-automation/cli-tests/tests/testHmy.py

```diff
@@ -42,77 +28,172 @@ def load_environment():
-        sys.exit(-1)
+        LOGGER(f"Failed: Could not get keystore path.\n"
+               f"\tGot exit code {err.returncode}. Msg: {err.output}")
+        return False
     if not os.path.exists(response):
-        log(f"[Critical] '{response}' is not a valid path")
-        sys.exit(-1)
+        LOGGER(f"Failed: '{response}' is not a valid path")
+        return False
     KEYSTORE_PATH = response
+    LOGGER("Passed", error=False)
+    return True
 
 
-def delete_from_keystore_by_name(name):
-    log(f"[KEY DELETE] Removing {name} from keystore at {KEYSTORE_PATH}", error=False)
-    key_file_path = f"{KEYSTORE_PATH}/{name}"
+@test_announce
+def test_and_load_keys_list():
+    """
+    CRITICAL TEST
+    """
+    try:
+        load_addresses()
+    except RuntimeError as err:
+        LOGGER(f"Failed: got error: {err}")
+        return False
+    LOGGER("Passed", error=False)
+    return True
+
+
+@test_announce
+def test_balance():
+    """
+    CRITICAL TEST
+
+    Currently only checks that s0 endpoint balance matches.
+      - Assumes that s1 balance works iff s0 balance works
+    """
+    with open("testHmyReferences/balance.json") as file:
+        balance_ref = json.load(file)
+    ref_key = balance_ref["key"]
+    ref_min_bal = balance_ref["min_balance"]["shard_0"]
+    url = 'http://localhost:9500/'
+    payload = "{\n    \"jsonrpc\": \"2.0\",\n    \"method\": \"hmy_getBalance\",\n    \"params\": " \
+              "[\n        \"" + ref_key + "\",\n        \"latest\"\n    ],\n    \"id\": 1\n}"
+    headers = {
+        'Content-Type': 'application/json'
+    }
 
     try:
-        shutil.rmtree(key_file_path)
-    except shutil.Error as e:
-        log(f"[KEY DELETE] Failed to delete dir: {key_file_path}\n"
-            f"Exception: {e}")
-        return
+        cli_response = subprocess.check_output(["hmy", "balance", ref_key], env=ENVIRONMENT).decode().strip()
+    except subprocess.CalledProcessError as err:
+        LOGGER(f"Failed: Could not get balance.\n"
+               f"Got exit code {err.returncode}. Msg: {err.output}")
+        return False
+    try:
+        cli_response_list = eval(cli_response.strip())
+    except SyntaxError:
+        LOGGER(f"Failed: Unexpected format of cli_response. Got: {cli_response}")
+        return False
 
-    del ADDRESSES[name]
+    response = requests.request('POST', url, headers=headers, data=payload, allow_redirects=False, timeout=3)
+    body = json.loads(response.content)
+    request_bal = round(int(body["result"], 16) * 10 ** -18, 6)
 
+    if ref_min_bal > request_bal:
+        LOGGER(f"Failed: Balance for reference is {request_bal} but need at least {ref_min_bal} for test to be valid.")
+        return False
 
-def get_address_from_name(name):
-    if name in ADDRESSES:
-        return ADDRESSES[name]
-    else:
-        load_addresses()
-        return ADDRESSES.get(name, None)
+    cli_s0_bal = round(cli_response_list[0]["amount"], 6)
```

Comment:
double check the determinism of this rounding logic, fix in upstream hmy CLI if need be